pyccel/codegen/printing/fcode.py

```diff
@@ -954,7 +954,8 @@ def _print_NumpyLinspace(self, expr):
         else:
             template = '[(({start} + {index}*{step}), {index} = {zero},{end})]'
             var = Variable('int', 'linspace_index')
-            self.scope.insert_variable(var)
+            if var.name not in self.scope.variables:
```

Comment:
I am not sure why this is needed. Is it because the name is not being chosen by the scope (as described in the developer docs) leading to name collisions?

Reply:
The name should be chosen by the scope to avoid collisions. If you want to reuse the variable then you should collect it from the scope not create a new one. I don't think it's worth putting this in this PR though. I plan to adapt the function `get_temporary_variable` to allow reuse at some point, so if you use this then I should find it when I get round to that

Reply:
Exactly, do you think it is better to get a new suffixed variable for each linspace declaration? Since it is a temporary variable, I think just one will do.

Reply:
This is a good point. I think that we should reuse the same temporary variable if it already exists in the scope, but how can we be sure that `linspace_index` was not created by the user for another purpose?

Reply:
-- as usual GitHub does not update the conversation in real time, so the comments are not reported in the correct order --

Reply:
We cannot be sure which is why we must use either:
- `get_new_temporary_variable` to create a temporary
- `get_new_name` + `Variable`

However the problem is that we then cannot be certain of the variable name on the second pass which makes it hard to search for it in the scope.

That's why I prefer using `get_new_temporary_variable` to create a `Dummy_XXX` Variable that hopefully one day we will be able to reuse in various contexts not just in the linspace index

Reply:
@EmilyBourne I agree with you, we should indeed use `get_new_temporary_variable` here.

I suppose the same bug appears in several other places across Pyccel....

Reply:
Noted, I will use `get_new_temporary_variable` instead.

Reply:
is [this](https://github.com/pyccel/pyccel/pull/1319/commits/7dbc6c4f8333f95a64ba835c306cd99e34fcbad8) better?

Reply:
Again, it looks good to me 😅 

@EmilyBourne I've just noticed that the function is called `get_temporary_variable` (without `new`). Such a name suggests that an existing temporary variable could be provided if available, although that is not the case.

Reply:
Ah sorry. I checked the name and still got it wrong! That impression is intended. I hope this function could be tweaked to allow that in the future. I should create an issue to describe this